app/types/nonCr3.ts

```diff
@@ -0,0 +1,86 @@
+import { z } from "zod";
+
+/**
+ * This is a 4x4 decimal degree bounding box centered on the tx state
+ * capitol building
+ * */
+const ATX_BBOX = {
+  latitude: {
+    min: 28.2747,
+    max: 32.2747,
+  },
+  longitude: {
+    min: -99.7404,
+    max: -95.7404,
+  },
+};
+
+/**
+ * Validation schema for a non-CR3 record uploaded through the UI
+ */
+export const NonCr3UploadSchema = z.object({
+  date: z.string().regex(/^\d{4}-\d{2}-\d{2}$/, "Date is missing or invalid"),
+  case_id: z.string().regex(/^\d+$/, "Case ID is missing or invalid"),
+  address: z
+    .string()
+    /**
+     * Remove all characters except letters, digits, whitespace, and these: `\/-.,&`.
+     */
+    .transform((val) => val.replace(/[^A-Za-z0-9\\/\-\s.,&]/g, ""))
+    .refine((value) => value.trim() !== "", {
+      message: "Address is missing or invalid",
+    }),
+  longitude: z.coerce
+    .number()
+    .min(ATX_BBOX.longitude.min, {
+      message: `Longitude is less than the minimum bounds (${ATX_BBOX.longitude.min})`,
+    })
+    .max(ATX_BBOX.longitude.max, {
+      message: `Longitude is greater than the maximum bounds (${ATX_BBOX.longitude.max})`,
+    }),
+  latitude: z.coerce
+    .number()
+    .min(ATX_BBOX.latitude.min, {
+      message: `Latitude is less than the maximum bounds (${ATX_BBOX.latitude.min})`,
+    })
+    .max(ATX_BBOX.latitude.max, {
+      message: `Latitude is greater than the maximum bounds (${ATX_BBOX.latitude.max})`,
+    }),
+  hour: z.coerce.number().int().min(0).max(23, "Hour must be between 0 and 23"),
+});
+
+/**
+ * Type for a non-CR3 record uploaded through the UI -
+ * this type is inferred from the Zod schema
+ */
+export type NonCr3Upload = z.infer<typeof NonCr3UploadSchema>;
+
+/**
+ * Validation function which checks for duplicate `case_ids`
+ */
+const noDuplicateCaseIds = (array: NonCr3Upload[]) => {
+  const caseIds = new Set();
+  return !array.some((item) => {
+    if (caseIds.has(item.case_id)) {
+      return true;
+    }
+    caseIds.add(item.case_id);
+    return false;
+  });
+};
+
+/**
+ * Adds a dupe case ID test to the NonCr3Upload schema
+ */
+export const NonCr3UploadDedupedSchema = z
+  .array(NonCr3UploadSchema)
+  .refine(noDuplicateCaseIds, "Duplicate case IDs (case_id) detected");
```

Comment:
I don't know how common this problem is, but it was in the old VZE so i went ahead and added the test here. Without this check, Hasura is going to bomb with a very unhelpful error.